### PR TITLE
Hide Cmdr window on click outside of window

### PR DESCRIPTION
**Pull request by TsjipTsjip**
This is still quite rudimentary and needs some updates to allow for configuration from outside of the Window.lua script.

### CODE REVIEW

File: Cmdr/CmdrClient/CmdrInterface/Window.lua [hunk 1]
```diff
@@ -243,7 +243,16 @@ function Window:BeginInput(input, gameProcessed)
...
+		local ps = input.Position
+		local xPos = Gui.AbsolutePosition.X
+		local yPos = Gui.AbsolutePosition.Y
+		local xOff = Gui.AbsoluteSize.X
+		local yOff = Gui.AbsoluteSize.Y
+		if not(ps.X >= xPos and ps.X <= xPos+xOff and ps.Y >= yPos and ps.Y <= yPos+yOff) then
+			self:Hide()
+		end
+	elseif input.KeyCode == Enum.KeyCode.Down then -- Auto Complete Down
 		self:SelectVertical(1)
```
Comment (evaera) [inline Cmdr/CmdrClient/CmdrInterface/Window.lua:256]: Please space infix operators correctly as inline with the style of the rest of the project

Comment (TsjipTsjip) [reply Cmdr/CmdrClient/CmdrInterface/Window.lua]: I've spaced out the plus operators as requested. If I didn't do the inline bit correctly I'm afraid I don't quite understand what needs to happen. Still keeping this as a draft, as I'm looking to allow the user to configure this.

**Comment by TsjipTsjip** [comment]
Ready for review, I've implemented the configurable option. Two notes on this:
- I myself don't really like the name I've given to the option as I feel it misses the point. (Frames can't be focussed on, so speaking of focus is incorrect.) Rename this at will, I believe your vision on this would be better to keep the project more coherent.
- I did not update CHANGELOG.md, as this would create a conflict that has to be resolved. In addition, #129 is already on this file for the version command. This needs resolution before this branch is merged.

File: Cmdr/CmdrClient/CmdrInterface/Window.lua [hunk 2]
```diff
@@ -243,7 +243,14 @@ function Window:BeginInput(input, gameProcessed)
 		return
 	end
 
-	if input.KeyCode == Enum.KeyCode.Down then -- Auto Complete Down
+	if self.Cmdr.HideOnLostFocus and table.find({Enum.UserInputType.MouseButton1,Enum.UserInputType.MouseButton2,Enum.UserInputType.Touch},input.UserInputType) then
```
Comment (evaera) [inline Cmdr/CmdrClient/CmdrInterface/Window.lua:246]:
```suggestion
	if self.Cmdr.HideOnLostFocus and table.find({Enum.UserInputType.MouseButton1,Enum.UserInputType.MouseButton2,Enum.UserInputType.Touch},input.UserInputType) then
```

Should be spaces after commas, also I'd recommend moving this table out into a constant value in SNAKE_CASE

File: Cmdr/CmdrClient/CmdrInterface/Window.lua [hunk 3]
```diff
@@ -243,7 +244,7 @@ function Window:BeginInput(input, gameProcessed)
 		return
 	end
 
-	if self.Cmdr.HideOnLostFocus and table.find({Enum.UserInputType.MouseButton1,Enum.UserInputType.MouseButton2,Enum.UserInputType.Touch},input.UserInputType) then
+	if self.Cmdr.HideOnLostFocus and table.find(MOUSE_TOUCH_ENUM,input.UserInputType) then
```
Comment (evaera) [inline Cmdr/CmdrClient/CmdrInterface/Window.lua:247]:
```suggestion
	if self.Cmdr.HideOnLostFocus and table.find(MOUSE_TOUCH_ENUM, input.UserInputType) then
```

File: Cmdr/CmdrClient/CmdrInterface/Window.lua [hunk 4]
```diff
@@ -8,6 +8,7 @@ local Player = Players.LocalPlayer
 
 local LINE_HEIGHT = 20
 local WINDOW_MAX_HEIGHT = 300
+local MOUSE_TOUCH_ENUM = {Enum.UserInputType.MouseButton1,Enum.UserInputType.MouseButton2,Enum.UserInputType.Touch}
```
Comment (evaera) [inline Cmdr/CmdrClient/CmdrInterface/Window.lua:11]:
```suggestion
local MOUSE_TOUCH_ENUM = {Enum.UserInputType.MouseButton1, Enum.UserInputType.MouseButton2, Enum.UserInputType.Touch}
```

File: docs/api/CmdrClient.md [hunk 5]
```diff
@@ -44,6 +44,10 @@ docs:
       params:
         - name: isEnabled
           type: boolean
+    - name: SetHideOnLostFocus
```
Comment (evaera) [inline docs/api/CmdrClient.md:47]:
```suggestion
    - name: SetHideOnLostFocus
      since: NEXT
```

**Comment by evaera** [comment]
Please add an entry to CHANGELOG.md

**Comment by TsjipTsjip** [comment]
> Please add an entry to CHANGELOG.md

On it. Going to need to rebase my fork though, otherwise it'd cause a conflict.

**Comment by TsjipTsjip** [comment]
I also had to change the PR number at the version command to 130, was previously listed as 129.

**Comment by evaera** [comment]
Great thanks

**Pull request merged**